gruntwork-install

```diff
@@ -114,7 +114,7 @@ function fetch_script_module {
 
   # We want to make sure that all folders down to $download_path/$module_name exists, but that $download_path/$module_name itself is empty.
   mkdir -p "$download_dir/$module_name/"
-  rm -Rf "$download_dir/$module_name/"
+  rm -Rf "${download_dir:?}/${module_name:?}/"
```

Comment:
Why the curly braces? And what's the `:?` do?

Reply:
Here is the error message:

> Use "${var:?}" to ensure this never expands to / .

I didn't think about this until I read that, but it is basically protecting against accidentally `rm -rf /` when both `download_dir` and `module_name` are blank. `:?` causes the command to fail if the variable is undefined.

You can read more about it in https://github.com/koalaman/shellcheck/wiki/SC2115

Reply:
Oh... wow, OK, makes sense.